config/src/network_id.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub enum NetworkType {
+    Validator = 0,
+    Public,
+    Private,
```

Comment:
Why didn't we follow the proposed format of

```
pub enum NetworkType {
  Validator,
  Public,
  Private(PrivateNetwork),
}

pub struct PrivateNetwork {
  name: String,
}
```

Reply:
TOML serialization determines that an ENUM with any extra values was an UnsupportedType.  See: https://github.com/libra/libra/issues/4210

I can go back and try it, but this should get the ball rolling.

Reply:
For everyone's knowledge, toml, won't serialize the enum:
- toml: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0e206f00f9d42117e9c3dc1906ba1762
- yaml: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0e206f00f9d42117e9c3dc1906ba1762

Reply:
Got it mostly working after putting in a tag.  Now just some side effects in the serde analyzer.